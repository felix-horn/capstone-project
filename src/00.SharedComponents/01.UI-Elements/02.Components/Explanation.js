import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Explanation.propTypes = {
  useCase: PropTypes.string.isRequired,
  className: PropTypes.string,
}

export default function Explanation({ useCase }) {
  return (
    <LayoutWrapper data-testid="component">
      {useCase === 'uncheckItem' &&
        'Um einen Artikel wieder auf dessen Einkaufsliste zu setzen, scanne dessen Barcode.'}
      {useCase === 'setup' &&
        'Scanne den Barcode des Artikels einmalig ein, um diesen zukünftig über die Scanner-Funktion der App wieder auf dessen Liste zu setzen.'}
    </LayoutWrapper>
  )
}

const LayoutWrapper = styled.div`
  padding: 0 30px;
  display: grid;
  place-items: center;
  font-size: 0.8rem;
  color: var(--dark-gray);
`
