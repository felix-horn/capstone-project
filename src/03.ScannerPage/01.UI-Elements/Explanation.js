import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Explanation.propTypes = {
  className: PropTypes.string,
}

export default function Explanation({ className }) {
  return (
    <ExplanationStyled className={className}>
      Scanne den Barcode des Artikels einmalig ein, um diesen zukünftig über die
      Scanner-Funktion der App wieder auf dessen Liste zu setzen.
    </ExplanationStyled>
  )
}

const ExplanationStyled = styled.div`
  padding: 0 30px;
  display: grid;
  place-items: center;
  font-size: 0.8rem;
  color: var(--dark-gray);
`
