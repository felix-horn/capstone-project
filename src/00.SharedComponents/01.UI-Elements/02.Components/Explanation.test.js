import { render } from '@testing-library/react'
import Explanation from './Explanation'

describe('Explanation', () => {
  it('renders correctly', () => {
    const { container } = render(<Explanation useCase="setup" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('displays the setup text', () => {
    const { queryByText, getByText } = render(<Explanation useCase="setup" />)
    const setupText = getByText(/einmalig/i)
    const uncheckItemText = queryByText(/Einkaufsliste/i)
    expect(setupText).toBeInTheDocument()
    expect(uncheckItemText).not.toBeInTheDocument()
  })

  it('displays the uncheckItem text', () => {
    const { queryByText, getByText } = render(
      <Explanation useCase="uncheckItem" />
    )
    const setupText = queryByText(/einmalig/i)
    const uncheckItemText = getByText(/Einkaufsliste/i)
    expect(setupText).not.toBeInTheDocument()
    expect(uncheckItemText).toBeInTheDocument()
  })
})
