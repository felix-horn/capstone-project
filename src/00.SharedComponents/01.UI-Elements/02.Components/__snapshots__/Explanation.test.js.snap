// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explanation renders correctly 1`] = `
.c0 {
  padding: 0 30px;
  display: grid;
  place-items: center;
  font-size: 0.8rem;
  color: var(--dark-gray);
}

<div
  class="c0"
>
  Scanne den Barcode des Artikels einmalig ein, um diesen zukünftig über die Scanner-Funktion der App wieder auf dessen Liste zu setzen.
</div>
`;
